scripts/env_wrapper.sh

```diff
@@ -45,12 +59,49 @@ fi
 cmd=("$@")
 if should_use_lima; then
     setup_lima
-    cache_dir="/home/debian/mkosi-cache"
-    cache_cmd="mkdir -p \"$cache_dir\" || true"
+
+    mkosi_cache=/home/debian/mkosi-cache
+    mkosi_output=/home/debian/mkosi-output
+
     if [[ "${cmd[0]}" == "mkosi" ]]; then
-        cmd+=("--cache-directory=$cache_dir")
+        limactl shell "$LIMA_VM" mkdir -p "$mkosi_cache" "$mkosi_output"
+
+        cmd=(
+            # First off, we need to run mkosi in a new user namespace,
+            # because it creates files owned by multiple uids/gids, and we
+            # either need to run as root, or use this unshare, which is safer.
+            # See also:
+            # https://manpages.debian.org/unstable/mkosi/mkosi.1.en.html#:~:text=Why%20do%20I%20see%20failures%20to%20chown%20files%20when%20building%20images
+            "unshare"
+            "--map-auto" "--map-current-user"
+            "--setuid=0" "--setgid=0"
+            "--"
+            # Next, we use which because actual file is somewhere in /nix/store
+            # and it's easier to explicitly specify it here instead of passing
+            # $PATH.
+            '$(which mkosi)'
```

Comment:
Due to some nix quirks, we'll need to pass in $PATH for maximum reproducibility (in both the lima path and the non-lima path)